code/modules/urist/antagonist/revenant/archetypes/chimera.dm

```diff
@@ -101,51 +101,103 @@
 /mob/proc/bsrevenant_minor_heal()
 	set name = "Fleshmend"
 	set category = "Anomalous Powers"
-	set desc = "Heal minor damage, stop bleeding. Use multiple times per each wound."
-
-	BSR_ABORT_IF_DEAD_PRESET(src)
+	set desc = "Unnatural regeneration. Revive on death, heal wounds/bloodloss otherwise."
 
 	// if TRUE, we're still waiting for a handler
 	var/pending = TRUE
 
+	var/datum/bluespace_revenant/revenant = src?.mind?.bluespace_revenant
+	var/valid_revenant = istype(revenant)
+
 	var/mob/living/carbon/human/H = src
 
 	if(istype(H))
+		// First things first - if they are dead, revive
 		if(pending)
-			for(var/obj/item/organ/I in H.internal_organs)
-				if(I.damage >= 5)
-					I.damage = max(I.damage - 5, 0)
-					to_chat(H, SPAN_NOTICE("Your [I.name] itches as you knit its wounds shut."))
-					pending = FALSE
-					break
+			if(H.stat == DEAD)
+				H.adjustOxyLoss(-rand(15,20))
```

Comment:
You probably want to bump this up since anyone dead's going to have 200 oxyloss and will get pretty significant brain damage by the time they're awake.

Reply:
This value is actually stolen from Bay's psionics revive code. Keep in mind, this power is meant to potentially be used multiple times in a row - obviously, for a higher total cost in realityfuckery - so brain damage can be healed by that too.